frontend/online-shop/src/components/product-details/products-details.component.tsx

```diff
@@ -0,0 +1,56 @@
+import { Button, Typography } from "@mui/material";
+
+import { FC, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchProductById } from "../../services/products.service";
+
+const ProductDetail: FC = () => {
+  let params = useParams();
+
+  const [product, setProduct] = useState({
+    name: "",
+    description: "",
+    category: "",
+    price: "",
+    image: "",
+  });
+  // eslint-disable-next-line
+  const [shoppingCart, setShoppingCart] = useState([]);
+  const handleAddItemToCart = () => {
+    console.log("item added");
+  };
+  useEffect(() => {
+    fetchProductById({ setProduct, params });
```

Comment:
Why are you passing `setProduct` and `params` to the `fetchProductById` method? It should only receive the ID of the product which we want to fetch. So something like `fetchProductById(params.id)`